sample-apps/javascript-sample-app/server.js

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS'" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ */
+
+'use strict';
+
+// setting up metrics
+// const my_meter = require('./create-a-meter');
+// const { emitsPayloadMetric, emitReturnTimeMetric } = require('./get-meter-emit-functions')(my_meter)
+
+// TracerProvider must be initialized before instrumented packages
+// (i.e. before 'aws-sdk' and 'http') are imported.
+
+// setting up the traces and metrics
+const tracer = require('./tracer');
+const meter = require('./meter');
+
+const http = require('http');
+const AWS = require('aws-sdk');
+const Worker = require('worker_threads');
+
+const { updateTotalBytesSent, updateLatencyTime } = require('./request-metrics')(meter);
```

Comment:
It is the first time that I see this. is this an usual construct? If yes, I'm ok with it, but if no, I'd prefer to play simple and have a class defined in request-metrics and you can pass the meter as parameter to the constructor.

Also meter is passed as parameter to request-metrics while we generate a new meter inside random-metrics. is it possible to be consistent?

Reply:
I have seen it before, it utilizes [JS ES6 Arrow function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions). It's essentially an anonymous function.

Fixed the second.